### DISCUSSION TITLE
It appears Enums cannot be used in ConfigurableResources. Is this intended?

### DISCUSSION THREAD

**Discussion by easontm** [dagster-io/dagster#18028] [Q&A]
Below I have a self-contained example which will show the behavior I'm observing. The main thing we are concerned with is the default value of `MyConfigResource.val`. There is nothing I can put here that will result in a successful execution.
```python3
from enum import Enum
from dagster import ConfigurableResource, op, job
from pydantic import Field

class MyEnum(Enum):
    one = "1"
    
class MyConfigResource(ConfigurableResource):
    val: MyEnum = "one"

@op
def myop(my_config_resource: MyConfigResource):
    print(my_config_resource.val)

@job(resource_defs={"my_config_resource": MyConfigResource()})
def myjob():
    myop()

if __name__ == "__main__":
    myjob.execute_in_process()
```

Here is what I observed when putting values in:

- `"one"` - The job tries to run but I get a validation error:
    ```
    pydantic_core._pydantic_core.ValidationError: 1 validation error for MyConfigResource
    val
      Input should be '1' [type=enum, input_value='one', input_type=str]
    ```
- `MyEnum.one` - I get a pre-execution validation error
    ```
        raise DagsterInvalidConfigError(
    dagster._core.errors.DagsterInvalidConfigError: Invalid default_value for Field.
        Error 1: Value at the root for enum type MyEnum must be a string
    ```
- `"1"` - A similar pre-execution validation error
    ```
        raise DagsterInvalidConfigError(
    dagster._core.errors.DagsterInvalidConfigError: Invalid default_value for Field.
        Error 1: Value at the root not in enum type MyEnum got 1
    ```

The behavior is also exactly the same if I use a `Field`, i.e.
```python3
from pydantic import Field

...

class MyConfigResource(ConfigurableResource):
    val: MyEnum = Field(default="one")
```

Interestingly, if you wrap the enum class in another type (like `list`), one of the options works
```python3
class MyConfigResource(ConfigurableResource):
    val: list[MyEnum] = ["one"]    # works
    val: list[MyEnum] = [MyEnum.one]  # doesn't work
    val: list[MyEnum] = ["1"]  # also doesn't work
```

To be clear I really only think `MyEnum.one` should be accepted at all. I'm quite annoyed that when wrapping it in a `list` the one that works is `"one"` -- making me use the string literal representing the enum key is quite odd.

**Comment by alangenfeld** [comment]
Ya this is something that needs improvement, here are a few issues associated which contain some workaround discussions you could reference
https://github.com/dagster-io/dagster/issues/15123
https://github.com/dagster-io/dagster/issues/17004
